src/app/modules/main/chat_section/chat_content/messages/module.nim

```diff
@@ -401,10 +403,18 @@ method reevaluateViewLoadingState*(self: Module) =
                 self.view.getMessageSearchOngoing()
   self.view.setLoading(loading)
 
+proc applyPinToMessages*(self: Module, messageItems: var seq[Item]) =
+  for item in messageItems:
+    if self.pinnedMessages.hasKey(item.id):
```

Comment:
What about the opposite, when it had been unpinned?

Reply:
Done below in `onUnpinMessage`